qi-lib/flow/core/impl.rkt

```diff
@@ -199,15 +194,6 @@
   (λ args
     (apply values (zip-with call fs args))))
 
-(define (all? . args)
-  (and (for/and ([v (in-list args)]) v) #t))
-
-(define (any? . args)
-  (and (for/or ([v (in-list args)]) v) #t))
-
-(define (none? . args)
-  (not (for/or ([v (in-list args)]) v)))
```

Comment:
Commit 268a5fd says these were removed because they are in qi-lib/flow/extended/impl.rkt, but those implementations are different (they omit the boolean casting; I _think_ we prefer without it).

I _believe_ what actually happened is that commit 8e4338f2b removed their use in favor of folds; later, commit c4244bc turned those forms into macros. I think then commit 22d2cee6 re-introduced `~all?` as a Qi function (so it's now a Qi function instead of a macro, and it does double-duty serving `AND`). Similar for commit 8156eb0d and `~any?`, and commit c0531e75 and `~none?`.

Reply:
Truth is stranger than commit messages 🌠

Reply:
It would be nice if this was more traceable, though. Spelunking through commits to answer logs is helped by cross-references and detailed commit messages :)

Reply:
True, I'll try to write better commit messages and think more "archeologically" about this :)